File: algorithms/167. Two Sum II - Input array is sorted/solution3.py
class Solution(object):
    def twoSum(self, numbers, target):
        """
        :type numbers: List[int]
        :type target: int
        :rtype: List[int]
        """
        for i in xrange(len(numbers)):
            j = bisect.bisect_left(numbers, target - numbers[i], i + 1)
            if j < len(numbers) and numbers[j] == target - numbers[i]:
                return [i + 1, j + 1]

File: algorithms/167. Two Sum II - Input array is sorted/solution3.cc
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        for ( int i = 0; i < numbers.size(); i++ ) {
            int k = target - numbers[i];
            auto j = lower_bound(numbers.begin() + i + 1, numbers.end(), k);
            if ( j != numbers.end() && *j == k ) {
                return {i + 1, j - numbers.begin() + 1};
            }
        }
    }
};
